carbon-design-system/carbon-for-ibm-dotcom
# fix(content-block-horizontal): ensure border attribute returns correctly

### Related Ticket(s)

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/8348

### Description

- This PR ensures that when the `border` attribute/prop is defined in the `<dds-content-block-horizontal>` component that the border-bottom renders and is hidden when it's not defined. 
- See the issue for more detail of the current behavior seen by the AEM adopters - https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/8348

### Changelog

**Changed**

- Ensure that the `border` attribute is defined
- Set border default prop to `false`.

### Testing

- Visit the `Content Block Horizontal` layout component code sandbox
- Add `border` to the `<dds-content-block-horizontal>` element (eg. `<dds-content-block-horizontal border>` and ensure that the border-bottom renders correctly.
- If the `border` attribute is not being used, ensure that the border-bottom does not get rendered.
